java/src/jmri/jmrix/dccpp/DCCppPredefinedMeters.java

```diff
@@ -17,49 +24,96 @@
 
     private DCCppTrafficController tc = null;
     private final MeterUpdateTask updateTask;
-    private final Meter currentMeter;
+    private String systemPrefix = null;
+    private HashMap<String, Meter> meters = new HashMap<String, Meter>(2); //keep track of defined meters
 
     public DCCppPredefinedMeters(DCCppSystemConnectionMemo memo) {
-        
+        log.debug("Constructor called");
+
+        systemPrefix = memo.getSystemPrefix();
         tc = memo.getDCCppTrafficController();
 
-        updateTask = new MeterUpdateTask() {
+        updateTask = new MeterUpdateTask(3000, 10000) {
             @Override
             public void requestUpdateFromLayout() {
                 tc.sendDCCppMessage(DCCppMessage.makeReadTrackCurrentMsg(), DCCppPredefinedMeters.this);
             }
         };
-        
-        currentMeter = new DefaultMeter.DefaultCurrentMeter(
-                memo.getSystemPrefix() + "V" + "BaseStationCurrent",
-                Meter.Unit.Percent, 0, 100.0, 1.0, updateTask);
-        
-        InstanceManager.getDefault(MeterManager.class).register(currentMeter);
-        
+
         // TODO: For now this is OK since the traffic controller
         // ignores filters and sends out all updates, but
         // at some point this will have to be customized.
-        tc.addDCCppListener(DCCppInterface.THROTTLE, this);
+        tc.addDCCppListener(DCCppInterface.CS_INFO, this);
 
-        //is_enabled = false;
         updateTask.initTimer();
-
-        log.debug("DCCppPredefinedMeters constructor called");
+        updateTask.enable(); //enable the send even if no meters exist yet
     }
 
     public void setDCCppTrafficController(DCCppTrafficController controller) {
         tc = controller;
     }
 
+    /* handle new Meter replies and original current replies 
+     *   creates meters if first time this name is encountered
+     *   builds original "current percent" meter or as defined in 
+     *   new MeterReply message format                               */
     @Override
     public void message(DCCppReply r) {
-        if (r.isCurrentReply()) {
-            log.debug("DCCppPredefinedMeters received reply: '{}'", r);
-            try {
-                currentMeter.setCommandedAnalogValue(((r.getCurrentInt() * 1.0f) / (DCCppConstants.MAX_CURRENT * 1.0f)) * 100.0f );  // return as percentage.
-            } catch (JmriException e) {
-                log.error("exception thrown when set current", e);
+
+        //bail if other message types received
+        if (!r.isCurrentReply() && !r.isMeterReply()) return; 
+        
+        log.debug("Handling reply: '{}'", r);
+
+        //assume old-style current message and default name and settings
+        String meterName = "CurrentPct";
```

Comment:
What does `CurrentPct` stands for? `Current percent`? No big deal, but might be good to add a comment for other developers. But no need to do in this PR, it can wait if you want.

Reply:
done.